azure/services/resourceskus/sku.go

```diff
@@ -128,3 +130,23 @@ func (s SKU) GetCapability(name string) (string, bool) {
 	}
 	return "", false
 }
+
+// HasLocationCapability returns true if the provided resource supports the location capability.
+func (s SKU) HasLocationCapability(name, size, location string) bool {
+	if s.LocationInfo != nil && *s.Name == size {
+		for _, info := range *s.LocationInfo {
+			if info.ZoneDetails != nil && *info.Location == location {
+				for _, zone := range *info.ZoneDetails {
+					if zone.Capabilities != nil {
+						for _, capability := range *zone.Capabilities {
```

Comment:
this capability can be exposed in a subset of the zones in a region, should we check against the VM zone here? example:

```json
"locationInfo": [
      {
        "location": "southcentralus",
        "zoneDetails": [
          {
            "Name": [
              "3"
            ],
            "capabilities": [
              {
                "name": "UltraSSDAvailable",
                "value": "True"
              }
            ],
            "name": null
          }
        ],
        "zones": [
          "1",
          "2",
          "3"
        ]
      }
    ],
```

Reply:
i didn't quiet understand the check you are referring here, zoneDetails already contains the name of the zone in which the UltraSSD is available, so what we need to check ?

Reply:
where are we checking against the zone name? we are checking against location only as far as I see?

Reply:
we need to check zoneDetails.Names contains the zone we are deploying to